### DISCUSSION TITLE
finished abnormally, Err code: -15 (running as script did not help)

### DISCUSSION THREAD

**Discussion by swiesenfeld0** [ablab/spades#1025] [Q&A] [answered]
Hi, I'm trying to run SPAdes for the first time, and I'm getting these parameters & spades log.
I saw that there was a previous thread with err code -15, but the solution posted did not fix my problem. Also, running the toy data set does work on my computer.

#319 

If anyone knows what the problem is, I would greatly appreciate the help! Any ideas of things I could try to troubleshoot this would also be appreciated.

params.txt 
`Command line: /n/home05/swiesenfeld/miniconda3/bin/spades.py    --pe1-1 /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R1_paired.fq.gz      --pe1-2 /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R2_paired.fq.gz      --pe1-s /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R1_unpaired.fq.gz    --pe1-s /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R2_unpaired.fq.gz    --careful       -t      10      -o      /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/assembled_3318_2I

System information:
  SPAdes version: 3.13.0
  Python version: 3.9.12
  OS: Linux-3.10.0-1160.36.2.el7.x86_64-x86_64-with-glibc2.17

Output dir: /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/assembled_3318_2I
Mode: read error correction and assembling
Debug mode is turned OFF

Dataset parameters:
  Multi-cell mode (you should set '--sc' flag if input data was obtained with MDA (single-cell) technology or --meta flag if processing metagenomic dataset)
  Reads:
    Library number: 1, library type: paired-end
      orientation: fr
      left reads: ['/n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R1_paired.fq.gz']
      right reads: ['/n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R2_paired.fq.gz']
      interlaced reads: not specified
      single reads: ['/n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R1_unpaired.fq.gz', '/n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R2_unpaired.fq.gz']
      merged reads: not specified
Read error correction parameters:
  Iterations: 1
  PHRED offset will be auto-detected
  Corrected reads will be compressed
Assembly parameters:
  k: automatic selection based on read length
  Repeat resolution is enabled
  Mismatch careful mode is turned ON
  MismatchCorrector will be used
  Coverage cutoff is turned OFF
Other parameters:
  Dir for temp files: /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/assembled_3318_2I/tmp
  Threads: 10
  Memory limit (in Gb): 92
`

spades.log
`Command line: /n/home05/swiesenfeld/miniconda3/bin/spades.py    --pe1-1 /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R1_paired.fq.gz      --pe1-2 /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R2_paired.fq.gz      --pe1-s /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R1_unpaired.fq.gz    --pe1-s /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R2_unpaired.fq.gz    --careful       -t      10      -o      /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/assembled_3318_2I

System information:
  SPAdes version: 3.13.0
  Python version: 3.9.12
  OS: Linux-3.10.0-1160.36.2.el7.x86_64-x86_64-with-glibc2.17

Output dir: /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/assembled_3318_2I
Mode: read error correction and assembling
Debug mode is turned OFF

Dataset parameters:
  Multi-cell mode (you should set '--sc' flag if input data was obtained with MDA (single-cell) technology or --meta flag if processing metagenomic dataset)
  Reads:
    Library number: 1, library type: paired-end
      orientation: fr
      left reads: ['/n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R1_paired.fq.gz']
      right reads: ['/n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R2_paired.fq.gz']
      interlaced reads: not specified
      single reads: ['/n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R1_unpaired.fq.gz', '/n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R2_unpaired.fq.gz']
      merged reads: not specified
Read error correction parameters:
  Iterations: 1
  PHRED offset will be auto-detected
  Corrected reads will be compressed
Assembly parameters:
  k: automatic selection based on read length
  Repeat resolution is enabled
  Mismatch careful mode is turned ON
  MismatchCorrector will be used
  Coverage cutoff is turned OFF
Other parameters:
  Dir for temp files: /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/assembled_3318_2I/tmp
  Threads: 10
  Memory limit (in Gb): 92

======= SPAdes pipeline started. Log can be found here: /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/assembled_3318_2I/spades.log

===== Read error correction started.

== Running read error correction tool: /n/home05/swiesenfeld/miniconda3/share/spades-3.13.0-0/bin/spades-hammer /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/assembled_3318_2I/corrected/configs/config.info

  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b
  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/assembled_3318_2I/corrected/configs/config.info
  0:00:00.039     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 1
  0:00:00.039     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 92 Gb
  0:00:00.039     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset
  0:00:00.059     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33
  0:00:00.059     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]
  0:00:00.059     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes
     === ITERATION 0 begins ===
  0:00:00.147     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index
  0:00:00.148     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 16 files using 1 threads. This might take a while.
  0:00:00.167     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 100000
  0:00:00.167     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 30.6654 Gb
  0:00:00.167     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 4194304
  0:00:00.168   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R1_paired.fq.gz
  0:00:16.176   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 325080 reads
  0:00:29.704   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 650062 reads
  0:00:43.166   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 975358 reads
  0:00:56.377   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1300611 reads
  0:01:09.393   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1626193 reads
  0:01:22.574   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1951499 reads
  0:01:35.774   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2276804 reads
  0:02:55.039   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 4228352 reads
  0:03:07.095   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R2_paired.fq.gz
  0:05:46.403   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 8395129 reads
  0:06:11.665   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R1_unpaired.fq.gz
  0:06:12.994   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R2_unpaired.fq.gz
  0:06:14.361   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 9002796 reads processed
  0:06:14.363     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.
  0:06:34.326     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 47380452 kmers in total.
  0:06:34.326     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.
  0:06:37.831     4M / 616M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices
  0:06:48.676    24M / 616M  INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.
  0:06:50.607    24M / 616M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 21978792 bytes occupied (3.71103 bits per kmer).
  0:06:50.724    24M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order
  0:06:55.333   752M / 752M  INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.
  0:11:44.191   752M / 752M  INFO    General                 (main.cpp                  : 155)   Extracting clusters
  0:11:57.650   752M / 2G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 24092170
  0:11:57.651   388M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.
  0:11:57.966     1G / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R1_paired.fq.gz

== Error ==  system call for: "['/n/home05/swiesenfeld/miniconda3/share/spades-3.13.0-0/bin/spades-hammer', '/n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/assembled_3318_2I/corrected/configs/config.info']" finished abnormally, err code: -15

In case you have troubles running SPAdes, you can write to spades.support@cab.spbu.ru
or report an issue on our GitHub repository github.com/ablab/spades
Please provide us with params.txt and spades.log files from the output directory.`

**Comment by asl** [comment]
Does the problem reproduce with the latest SPAdes 3.15.5?

**Reply by swiesenfeld0** [reply]
Hi @asl ! I updated SPAdes to 3.15.5 and now this is the error message (--continue does not allow it to pass this unfortunately

`
== Error ==  system call for: "['/n/home05/swiesenfeld/miniconda3/envs/assembler/bin/spades-hammer', '/n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/assembled_3318_2I/corrected/configs/config.info']" finished abnormally, OS return value: -15
None
`

New params.txt:
`Command line: /n/home05/swiesenfeld/miniconda3/envs/assembler/bin/spades.py     --pe1-1 /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R1_paired.fq.gz      --pe1-2 /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R2_paired.fq.gz      --pe1-s /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R1_unpaired.fq.gz    --pe1-s /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R2_unpaired.fq.gz    --careful
       -t      10      -o      /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/assembled_3318_2I

System information:
  SPAdes version: 3.15.5
  Python version: 3.8.13
  OS: Linux-3.10.0-1160.36.2.el7.x86_64-x86_64-with-glibc2.10

Output dir: /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/assembled_3318_2I
Mode: read error correction and assembling
Debug mode is turned OFF

Dataset parameters:
  Standard mode
  For multi-cell/isolate data we recommend to use '--isolate' option; for single-cell MDA data use '--sc'; for metagenomic data use '--meta'; for RNA-Seq use '--rna'.
  Reads:
    Library number: 1, library type: paired-end
      orientation: fr
      left reads: ['/n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R1_paired.fq.gz']
      right reads: ['/n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R2_paired.fq.gz']
      interlaced reads: not specified
      single reads: ['/n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R1_unpaired.fq.gz', '/n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/output_3118_R2_unpaired.fq.gz']
      merged reads: not specified
Read error correction parameters:
  Iterations: 1
  PHRED offset will be auto-detected
  Corrected reads will be compressed
Assembly parameters:
  k: automatic selection based on read length
  Repeat resolution is enabled
  Mismatch careful mode is turned ON
  MismatchCorrector will be used
  Coverage cutoff is turned OFF
Other parameters:
  Dir for temp files: /n/holylfs05/LABS/hanage_lab/Lab/holyscratch01/sophia/merged_reads3118/assembled_3318_2I/tmp
  Threads: 10
  Memory limit (in Gb): 92`

**Reply by asl** [reply]
Will you please attach a complete `spades.log` file?

**Reply by swiesenfeld0** [reply]
@asl Yes, here it is! Thank you for helping

[spades.log](https://github.com/ablab/spades/files/9541379/spades.log)

**Comment by asl** [comment, marked as answer]
I'm seeing lots of strange things in the log. First of all you're running only using 1 threads:
```
  0:00:00.012     1M / 29M   INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 1
```
You might want to talk with your system administrator to ask how one could properly specify # of OpenMP threads as it seems you're running on some kind of cluster environment.

Next, it seems like the SPAdes job was killed from outside. You could probably check for your cluster / system log for more clues. E.g. whether some disk / cpu / time quota was exceeded or something similar.

**Reply by swiesenfeld0** [reply]
That was the issue, everything is running fine now. I was having trouble setting the number of threads, but now things assemble very quickly and very well. Thank you!!!
